Src/OSD/Unix/FileSystemPath.cpp

```
@@ -0,0 +1,146 @@
+ ** Software Foundation, either version 3 of the License, or (at your option)
+ ** any later version.
+ **
+ ** Supermodel is distributed in the hope that it will be useful, but WITHOUT
+ ** ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ ** FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ ** more details.
+ **
+ ** You should have received a copy of the GNU General Public License along
+ ** with Supermodel.  If not, see <http://www.gnu.org/licenses/>.
+ **/
+
+#include "FileSystemPath.h"
+#include "Util/Format.h"
+#include <string>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <pwd.h>
+#include <unistd.h>
+
+namespace FileSystemPath
+{
+    // Checks if a directory exists (returns true if exists, false if it doesn't)
+    bool PathExists(std::string fileSystemPath)
+    {
+        bool pathExists = false;
+        struct stat pathInfo;
+
+        if (stat(fileSystemPath.c_str(), &pathInfo) == 0 && S_ISDIR(pathInfo.st_mode))
+        {
+            pathExists = true;
+        }
+
+        return pathExists;
+
+    }
+
+    // Generates a path to be used by Supermodel files
+    std::string GetPath(fsPathType pathType)
+    {
+        std::string finalPath = "";
+        std::string homePath = "";
+        std::string strPathType = "";
+		struct passwd* pwd = getpwuid(getuid());
+
+        // Resolve pathType to string for later use
+        switch (pathType)
+        {
+            case Analysis:
+                strPathType = "Analysis";
+                break;
+            case Config:
+                strPathType = "Config";
+                break;
+            case Log:
+                strPathType = "Log";
+                break;
+            case NVRAM:
+                strPathType = "NVRAM";
+                break;
+            case Saves:
+                strPathType = "Saves";
+                break;
+            case Screenshots:

```

I would add a default case. And I think that case should print an error and bring down Supermodel. We don't have a great way of doing this right now but I think you could probably throw a fatal exception to kill the application. The reason for this is to guard against programmer errors. For example, if someone adds an enum case and forgets to update the switch statements. Some languages will guard against this but C++ doesn't do this.

---

What about using a null string as default case? That would translate to current directory in Windows and Linux (if Config is in current path) or ~/.supermodel or ~/.local/share/supermodel depending on which path exists.

What do you think? It's simpler than bringing down Supermodel on error.